probe/endpoint/ebpf.go

```diff
@@ -84,6 +84,31 @@ func isKernelSupported() error {
 		return fmt.Errorf("got kernel %s but need kernel >=4.4", release)
 	}
 
+	if strings.Contains(version, "Ubuntu") {
+		// Check for specific Ubuntu kernel versions with
+		// known issues.
+
+		// The vendor specific release part could look like:
+		// '<ABI number>.<upload number>-<flavour>' or
+		// '<ABI number>-<flavour>'
+		// See https://wiki.ubuntu.com/Kernel/FAQ
+
+		// cut away flavour
+		vendorStr := strings.Split(releaseParts[4], "-")[0]
+		// cut away upload number
+		abiNumberStr := strings.Split(vendorStr, ".")[0]
```

Comment:
I had to read the docs for `Split` to see what it did if the separator isn't found.
Why not just include the pattern for the ABI number in `releaseRegex`, i.e. replace `(.*)$` with `(\d*)(.*)$` ?